src/app/components/article/ExplainerForm.js

```diff
@@ -0,0 +1,174 @@
+          dbid
+          name
+        }
+        team {
+          dbid
+          slug
+        }
+      }
+    }
+  }
+`;
+
+const ExplainerForm = ({
+  article,
+  team,
+  onClose,
+  setFlashMessage,
+}) => {
+  const type = article?.dbid ? 'edit' : 'create';
+  const [saving, setSaving] = React.useState(Boolean(false));
+  const [error, setError] = React.useState(Boolean(false));
+  const [explainer, setExplainer] = React.useState({
+    title: article?.title || null,
+    description: article?.descripton || null,
+    language: article?.language || null,
+    tags: article?.tags || [],
+    url: article?.url || null,
+  });
+
+  const handleSuccess = () => {
+    setSaving(false);
+    setError(false);
+    onClose(false);
+    setFlashMessage(<FormattedMessage
+      id="explainerForm.saveSuccess"
+      defaultMessage="Article created successfully!"
+      description="Save item action success message"
+    />, 'success');
+  };
+
+  const handleError = () => {
+    setSaving(false);
+    setError(false);
+    onClose(false);
+    setFlashMessage(<FormattedMessage
+      id="explainerForm.saveError"
+      defaultMessage="Error with saving your article"
+      description="save item action failure message"
+    />, 'error');
+  };
+
+  const handleSave = () => {
+    setSaving(true);
+    commitMutation(Relay.Store, {
+      mutation: createMutation,
+      variables: {
+        input: {
+          ...explainer,
+        },
+      },
+      onCompleted: (response, err) => {
+        setSaving(false);
+        if (err) {
+          handleError(err);
+        } else {
+          handleSuccess(response);
+        }
+      },
+      onError: () => {
+        setSaving(false);
+        setError(true);
+      },
+    });
+  };
+
+  const handleBlur = (field, value) => {
+    setExplainer({ ...explainer, [field]: value });
+    if (type === 'edit') {
+      setSaving(true);
+      commitMutation(Relay.Store, {
+        mutation: updateMutation,
+        variables: {
+          input: {
+            id: article.id,
+            ...explainer,
+          },
+        },
+        onCompleted: (response, err) => {
+          setSaving(false);
+          if (err) {
+            setError(true);
+          } else {
+            setError(false);
+          }
+        },
+        onError: () => {
+          setSaving(false);
+          setError(true);
+        },
+      });
+    }
+  };
+
+  return (
+    <>
+      <ArticleForm
+        handleSave={handleSave}
+        onClose={onClose}
+        handleBlur={handleBlur}
+        articleType="explainer"
+        mode={type}
+        article={article}
+        team={team}
+        saving={saving}
+        error={error}
+      />
+    </>
+  );
+};
+
+ExplainerForm.defaultProps = {
+  article: null,
+};
+
+ExplainerForm.propTypes = {
+  article: PropTypes.object,
+  team: PropTypes.object.isRequired,
+  onClose: PropTypes.func.isRequired,
+};
+
+export default withSetFlashMessage(ExplainerForm);
```

Comment:
Please get `setFlashMessage` with the `useContext` hook:

as in `const setFlashMessage = React.useContext(FlashMessageSetterContext);`